GPL/Events/File/Probe.bpf.c

```diff
@@ -532,7 +543,151 @@ int BPF_KRETPROBE(kretprobe__chmod_common, int ret)
     if (!state)
         goto out;
 
-    chmod_common__exit(state->chmod.path, state->chmod.mode, ret);
+    chmod_common__exit(state->chmod.path, ret);
+
+out:
+    return 0;
+}
+
+SEC("kprobe/do_truncate")
+int BPF_KPROBE(kprobe__do_truncate)
+{
+    struct ebpf_events_state state = {};
+
+    struct file *filp;
+    if (FUNC_ARG_READ_PTREGS(filp, do_truncate, filp)) {
+        bpf_printk("kprobe__do_truncate: error reading filp\n");
+        return 0;
+    }
+
+    struct path path    = BPF_CORE_READ(filp, f_path);
+    state.truncate.path = &path;
+    ebpf_events_state__set(EBPF_EVENTS_STATE_TRUNCATE, &state);
+
+out:
+    return 0;
+}
+
+static int do_truncate__exit(struct path *path, int ret)
+{
+    if (ret)
+        goto out;
+
+    if (ebpf_events_is_trusted_pid())
+        goto out;
+
+    file_modify_event__emit(EBPF_FILE_CHANGE_TYPE_CONTENT, path);
+
+out:
+    return 0;
+}
+
+SEC("fexit/do_truncate")
+int BPF_PROG(fexit__do_truncate)
+{
+    struct file *filp = FUNC_ARG_READ(___type(filp), do_truncate, filp);
+    struct path path  = BPF_CORE_READ(filp, f_path);
+    int ret           = FUNC_RET_READ(___type(ret), do_truncate);
+    return do_truncate__exit(&path, ret);
+}
+
+SEC("kretprobe/do_truncate")
+int BPF_KRETPROBE(kretprobe__do_truncate, int ret)
+{
+    struct ebpf_events_state *state = ebpf_events_state__get(EBPF_EVENTS_STATE_TRUNCATE);
+    if (!state)
+        goto out;
+
+    do_truncate__exit(state->truncate.path, ret);
+
+out:
+    return 0;
+}
+
+SEC("kprobe/vfs_write")
+int BPF_KPROBE(kprobe__vfs_write, struct file *file, const char *buf, size_t count, loff_t *pos)
+{
+    struct ebpf_events_state state = {};
+
+    struct path path = BPF_CORE_READ(file, f_path);
+    state.write.path = &path;
+    ebpf_events_state__set(EBPF_EVENTS_STATE_WRITE, &state);
+
+    return 0;
+}
+
+static int vfs_write__exit(struct path *path, int ret)
+{
+    if (ret)
+        goto out;
+
+    if (ebpf_events_is_trusted_pid())
+        goto out;
+
+    file_modify_event__emit(EBPF_FILE_CHANGE_TYPE_CONTENT, path);
+
+out:
+    return 0;
+}
+
+SEC("fexit/vfs_write")
+int BPF_PROG(
+    fexit__vfs_write, struct file *file, const char *buf, size_t count, loff_t *pos, int ret)
+{
+    struct path path = BPF_CORE_READ(file, f_path);
+    return vfs_write__exit(&path, ret);
+}
+
+SEC("kretprobe/vfs_write")
+int BPF_KRETPROBE(kretprobe__vfs_write, int ret)
+{
+    struct ebpf_events_state *state = ebpf_events_state__get(EBPF_EVENTS_STATE_WRITE);
+    if (!state)
+        goto out;
+
+    vfs_write__exit(state->write.path, ret);
```

Comment:
```suggestion
    (void)vfs_write__exit(state->write.path, ret);
```
explicitly ignore the return? make `vfs_write__exit` a void function?